phabricatoremails/source.py

```diff
@@ -0,0 +1,73 @@
+import json
+from dataclasses import dataclass
+from typing import Optional
+
+import requests
+from requests import RequestException
+
+
+class PhabricatorException(Exception):
+    pass
+
+
+@dataclass
+class PhabricatorSource:
+    server: str
+    token: str
+    override_limit: Optional[int]
+
+    def _get(self, endpoint, **query_parameters):
+        query_parameters = {k: v for k, v in query_parameters.items() if v}
+        url = "{}/api/{}".format(self.server, endpoint)
+
+        body = {
+            "params": json.dumps(
+                {"__conduit__": {"token": self.token}, **query_parameters}
+            )
+        }
+
+        try:
+            result = requests.post(url, data=body)
+        except RequestException:
+            raise PhabricatorException("Could not communicate with Phabricator")
+
+        if result.status_code != 200:
+            raise PhabricatorException(
+                "Phabricator request had incorrect status of {}. "
```

Comment:
Is there a `format` missing here? You should probably be using f-strings since here and elsewhere where possible, since they are generally more readable.